devito/arch/compiler.py

```diff
@@ -414,7 +414,8 @@ def __init__(self, *args, **kwargs):
                 # -march isn't supported on power architectures
                 self.cflags += ['-mcpu=native']
             else:
-                self.cflags += ['-march=native']
+                pass
```

Comment:
first of all, I just noticed that within the outer `else` we have an `if platform in [POWER...] ... -else ...` that can probably be de-nested?

then, I would add:

```
elif platform in ['POWER8', 'POWER9']:
    self.cflags += ['-mcpu=native']
elif platform is ARM:
    self.libraries.remove('m')  # ugly, but as I said earlier, more refactoring otherwise necessary
else:
    self.cflags += ['-march=native']
```

Reply:
I think it's a bit trickier than that. We know the arm compiler flags are working correctly on a pure arm platform (i.e `-m`) this seems to be an `osx + arm` issue with apple shipping a compiler that has to holes with `clang`. 
So this may need to be slightly more subtle.